From 0e8ad6d22929faf6f47469df24cb0b4e96505c07 Mon Sep 17 00:00:00 2001
From: Ondrej Kozina <okozina@redhat.com>
Date: Fri, 19 Jun 2026 16:41:47 +0200
Subject: [PATCH 1/3] BACKPORT: Add keyring key type.

(cherry picked from commit b6fb6fc0fdc661afba734f4c29350fa6be2ede8f)
---
 lib/utils_keyring.c | 1 +
 lib/utils_keyring.h | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/utils_keyring.c b/lib/utils_keyring.c
index 73df3461be99..67e8db0498ac 100644
--- a/lib/utils_keyring.c
+++ b/lib/utils_keyring.c
@@ -32,6 +32,7 @@ static const struct {
 	{ BIG_KEY,       "big_key" },
 	{ TRUSTED_KEY,   "trusted" },
 	{ ENCRYPTED_KEY, "encrypted" },
+	{ KEYRING_KEY,	 "keyring" },
 };
 
 #include <linux/keyctl.h>
diff --git a/lib/utils_keyring.h b/lib/utils_keyring.h
index 67debf62a946..20c08293b52b 100644
--- a/lib/utils_keyring.h
+++ b/lib/utils_keyring.h
@@ -17,7 +17,7 @@
 typedef int32_t key_serial_t;
 #endif
 
-typedef enum { LOGON_KEY = 0, USER_KEY, BIG_KEY, TRUSTED_KEY, ENCRYPTED_KEY, INVALID_KEY } key_type_t;
+typedef enum { LOGON_KEY = 0, USER_KEY, BIG_KEY, TRUSTED_KEY, ENCRYPTED_KEY, KEYRING_KEY, INVALID_KEY } key_type_t;
 
 const char *key_type_name(key_type_t ktype);
 key_type_t key_type_by_name(const char *name);

base-commit: 738e7001cac91c555cd6798cca9194bb19e54e96
-- 
2.52.0

